metadata-ingestion/src/datahub/ingestion/api/source.py

```diff
@@ -400,13 +400,8 @@ class Source(Closeable, metaclass=ABCMeta):
     ctx: PipelineContext
 
     @classmethod
-    def create(cls, config_dict: dict, ctx: PipelineContext) -> Self:
-        # Technically, this method should be abstract. However, the @config_class
-        # decorator automatically generates a create method at runtime if one is
-        # not defined. Python still treats the class as abstract because it thinks
-        # the create method is missing. To avoid the class becoming abstract, we
-        # can't make this method abstract.
-        raise NotImplementedError('sources must implement "create"')
+    @abstractmethod
+    def create(cls, config_dict: dict, ctx: PipelineContext) -> Self: ...
```

Comment:
this is not a concern anymore because of the added `abc.update_abstractmethods(cls)` ?

Reply:
yup that's the idea. but abc.update_abstractmethods was only added in python 3.10, so I might need to revert this